performance/performance_tuning.rst

```diff
@@ -28,6 +28,155 @@ Hash based distribution is more suited to cases where users want to do real-time
 
+.. _examining_plan:
+
+Examining Distributed Query Plan
+##################################
+
+Citus extends the SQL EXPLAIN command to provide information about simple distributed queries, meaning queries that do not contain re-partition jobs. The EXPLAIN output consists of two parts: the distributed query and the master query. These parts provide information to help tune Citus and the underlying PostgreSQL database.
+
+Here is an example of explaining the plan for a query in the :ref:`hash partitioning tutorial <tut_hash>`:
+
+::
+
+  explain SELECT comment FROM wikipedia_changes c, wikipedia_editors e
+          WHERE c.editor = e.editor AND e.bot IS true LIMIT 10;
+
+  Distributed Query into pg_merge_job_0005
+    Executor: Real-Time
+    Task Count: 16
+    Tasks Shown: One of 16
+    ->  Task
+      Node: host=localhost port=9701 dbname=j
+      ->  Limit  (cost=0.15..6.87 rows=10 width=32)
+        ->  Nested Loop  (cost=0.15..131.12 rows=195 width=32)
+          ->  Seq Scan on wikipedia_changes_102024 c  (cost=0.00..13.90 rows=390 width=64)
+          ->  Index Scan using wikipedia_editors_editor_key_102008 on wikipedia_editors_102008 e  (cost=0.15..0.29 rows=1 width=32)
+            Index Cond: (editor = c.editor)
+            Filter: (bot IS TRUE)
+  Master Query
+    ->  Limit  (cost=0.00..0.00 rows=0 width=0)
+      ->  Seq Scan on pg_merge_job_0005  (cost=0.00..0.00 rows=0 width=0)
+  (15 rows)
+
+Let's examine it piece by piece. The first thing it reveals is the :ref:`distributed_query_executor` type being used and the number of shards (sixteen) that the query is referencing.
+
+::
+
+  Executor: Real-Time
+  Task Count: 16
+  Tasks Shown: One of 16
+
+Next explain goes into each shard of each worker and finds the plan used internally there by PostgreSQL. By default it shows the output from only one shard. In the case of this example it was running on a worker on localhost, port 9701.
+
+::
+
+  ->  Task
+    Node: host=localhost port=9701 dbname=j
+    ->  Limit  (cost=0.15..6.87 rows=10 width=32)
+      ->  Nested Loop  (cost=0.15..131.12 rows=195 width=32)
+        ->  Seq Scan on wikipedia_changes_102024 c  (cost=0.00..13.90 rows=390 width=64)
+        ->  Index Scan using wikipedia_editors_editor_key_102008 on wikipedia_editors_102008 e  (cost=0.15..0.29 rows=1 width=32)
+          Index Cond: (editor = c.editor)
+          Filter: (bot IS TRUE)
+
+The above is standard PostgreSQL explain output. You can use it for :ref:`postgresql_tuning` on the workers. After the output of distributed tasks there is a section about Citus' behavior on the master node.
+
+::
+
+  Master Query
+    ->  Limit  (cost=0.00..0.00 rows=0 width=0)
+      ->  Seq Scan on pg_merge_job_0005  (cost=0.00..0.00 rows=0 width=0)
+
+This explains how Citus is combining the results of each task to create the final result. If you see an expensive query plan in this area it means you should apply :ref:`distributed_query_performance_tuning` to improve it.
```

Comment:
The last line isn't entirely accurate, so let's take this out. The query plan alone won't give users information on how to tune their distributed settings.